mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -71,6 +71,26 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
 
   /** @group getParam */
   def getItemCol: String = $(itemCol)
+
+  /**
+   * Param for strategy for dealing with unknown or new users/items at prediction time.
+   * This may be useful in cross-validation or production scenarios, for handling user/item ids
+   * the model has not seen in the training data.
+   * Supported values:
+   * - "nan": predicted value for unknown ids will be NaN.
```

Comment:
It _might_ be worth it to make it cae insensitive since NaN is more commonly typed as NaN (and the other abbreviations used for algorithms are more commonly lower case).